setup.py

```diff
@@ -7,31 +7,43 @@
 )
 import subprocess
 
-test_deps = [
-    'pytest>=5.2.0,<6',
-    'pytest-cov>=2.8.1,<3',
-    'coveralls[yaml]>=1.8.2,<2',
-    'pytest-xdist>=1.30.0,<2',
-    'eth-tester[py-evm]>=0.3.0b1,<0.4',
-    'web3>=5.2.0,<5.3.0',
-    'tox>=3.7,<4',
-    'lark-parser>=0.7.8,<1',
-    'hypothesis[lark]>=4.53.2,<5',
-]
-lint_deps = [
-    'flake8>=3.7.9,<3.8.0',
-    'flake8-bugbear>=19.8.0,<20',
-    'flake8-use-fstring>=1.0.0,<2.0.0',
-    'isort>=4.2.15,<5',
-    'mypy>=0.740,<1',
-]
-
-
-extras = {
-    'test': test_deps,
-    'lint': lint_deps,
+extras_require = {
+    'test': [
+        'pytest>=5.2.0,<6',
+        'pytest-cov>=2.8.1,<3',
+        'coveralls[yaml]>=1.8.2,<2',
+        'pytest-xdist>=1.30.0,<2',
+        'eth-tester[py-evm]>=0.3.0b1,<0.4',
+        'web3>=5.2.0,<5.3.0',
+        'tox>=3.7,<4',
+        'lark-parser>=0.7.8,<1',
+        'hypothesis[lark]>=4.53.2,<5',
+    ],
+    'lint': [
+        'flake8>=3.7.9,<3.8.0',
+        'flake8-bugbear>=19.8.0,<20',
+        'flake8-use-fstring>=1.0.0,<2.0.0',
+        'isort>=4.2.15,<5',
+        'mypy>=0.740,<1',
+    ],
+    'docs': [
+        'recommonmark',
+        'sphinx',
+        'sphinx_rtd_theme',
+    ],
+    'dev': [
+        'ipython',
```

Comment:
it's a useful tool, but `ipython` as a dependency seems weird

Reply:
nothing under `dev` is a required dependency, just helpful